src/System.Composition.Runtime/tests/CompositionContextTests.cs

```diff
@@ -0,0 +1,284 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Composition.Hosting;
+using System.Composition.Hosting.Core;
+using Xunit;
+
+namespace System.Composition.Tests
+{
+    public class CompositionContextTests
+    {
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void GetExportT_Default_ReturnsExpected(bool success)
+        {
+            var context = new SubContext(contract =>
+            {
+                Assert.Equal(typeof(int), contract.ContractType);
+                Assert.Null(contract.ContractName);
+                Assert.Null(contract.MetadataConstraints);
+
+                return (success, 10);
+            });
+            if (success)
+            {
+                Assert.Equal(10, context.GetExport<int>());
+            }
+            else
+            {
+                Assert.Throws<CompositionFailedException>(() => context.GetExport<int>());
+            }
+        }
```

Comment:
It feels like a lot of these tests share the same structure and just have minor differences between them.  Is it possible to condense these via a helper that contains most of the boilerplate, or are they different enough that doesn't work?